### DISCUSSION TITLE
How to use Nested flows

### DISCUSSION THREAD

**Discussion by jyotipmahes** [NVIDIA-NeMo/Guardrails#65] [General]
Hi, I am looking for use cases where one flow can call another flow. How can we do that? Will it be supported in the future?

**Comment by drazvan** [comment]
Hi @jyotipmahes!

The documentation provides a simple example: https://github.com/NVIDIA/NeMo-Guardrails/blob/main/docs/user_guide/colang-language-syntax-guide.md#subflows. You can encapsulate a user authentication logic for example in a flow that gets called from another one. There are a few more examples in the tests section (https://github.com/NVIDIA/NeMo-Guardrails/blob/main/tests/test_subflows.py). We'll be adding more examples soon. Is there any specific use case you have in mind for using nested flows?

**Reply by jyotipmahes** [reply]
@drazvan Thanks for sharing this. I will try to use the above to see if this meets the need. I am just prototyping as of now so no specific use case

**Reply by lawya-cm** [reply]
Is it possible to route to different subflows conditionally within a single flow

e.g

    define flow emergency
        execute log_emergency(prompt=$last_user_message)
        when user fire
            do emergency fire
        when flood
            do emergency flood
        
        bot get location

    define bot fire response
        "If there is a fire call 911 immediately"

    define bot get location
        "What is the location of the emergency?"

    define bot fire extinguish
        "You should have a fire extinguisher under the kitchen sink"

    define subflow emergency fire
        bot fire response
        bot fire extinguish

    etc.

btw. the above does not work
